internal/agent/client/management.go

```diff
@@ -79,3 +101,19 @@ func (m *management) GetRenderedDevice(ctx context.Context, name string, params
 
 	return nil, resp.StatusCode(), nil
 }
+
+func PatchDeviceStatusSystemInfo(ctx context.Context, client Management, name string, systemInfo v1alpha1.DeviceSystemInfo) error {
+	data, err := util.StructToMap(systemInfo)
+	if err != nil {
+		return fmt.Errorf("convert system info to map: %w", err)
+	}
+	var value interface{} = data
+	patch := v1alpha1.PatchRequest{
+		{Op: "replace", Path: "/status/systemInfo", Value: &value},
+	}
+	if err := client.PatchDeviceStatus(ctx, name, patch); err != nil {
+		return fmt.Errorf("patch device status: %w", err)
+	}
+
+	return nil
```

Comment:
We could just return right away
```suggestion
	return client.PatchDeviceStatus(ctx, name, patch)
```